packages/teams-js/src/public/video.ts

```diff
@@ -125,6 +125,107 @@ export namespace video {
     sendMessageToParent('video.registerForVideoFrame', [config]);
   }
 
+  /**
+   * MediaStream response
+   */
+  export interface MediaStreamResponse {
+    // getTextureStream: ({streamId: string}) => Promise<MediaStream>;
+    /**
+     * The raw MediaStream
+     */
+    mediaStream: MediaStream;
+    // registerTextureStream: ({streamId: string, outputStreamTrack: MediaStreamTrack}) => Promise<void>;
+    /**
+     * register the processed track to te media stream
+     */
+    registerOutputStreamTrack: (outputStreamTrack: MediaStreamTrack) => void;
+
+    /**
+     * Get metadata of the video frame calulated by the native modules
+     */
+    getVideoFrameMetaData: (timestamp: number) => { [key: string]: ArrayBuffer };
+  }
+
+  type IPCInfoT2 = {
+    streamId: string;
+  };
+
+  /**
+   * get video stream in Permissions section
+   * @beta
+   */
+  export function getVideoStream(config: VideoFrameConfig): Promise<MediaStreamResponse> {
+    ensureInitialized(FrameContexts.sidePanel);
+    if (!isSupported()) {
+      throw errorNotSupportedOnPlatform;
+    }
+
+    // reuse the event name in registerForVideoFrame
+    sendMessageToParent('video.registerForVideoFrame', [config]);
+
+    // When getTextureStream is available, we will use it to get the video stream
+    // otherwise, wrap video frames to a MediaStream
+    if (window.chrome.webview.getTextureStream && window.chrome.webview.registerTextureStream) {
+      return new Promise((resolve, reject) => {
+        registerHandler(
+          // new event here
+          'video.startVideoExtensibilityVideoStream',
+          async (ipcInfo: IPCInfoT2) => {
+            if (ipcInfo) {
+              const { streamId } = ipcInfo;
+              // todo: error handling
+              const mediaStream = await window.chrome.webview.getTextureStream(streamId);
+              const mediaStreamResponse: MediaStreamResponse = {
+                mediaStream,
+                registerOutputStreamTrack: (outputStreamTrack: MediaStreamTrack) => {
+                  // TODO: calculate fps
+                  window.chrome.webview.registerTextureStream(streamId, outputStreamTrack);
+                },
+                getVideoFrameMetaData: (timestamp: number) => {
+                  // TODO: add getVideoFrameMetaData: how does the metadata get to the host?
```

Comment:
[CoreWebView2.PostWebMessageAsJson(String) Method (Microsoft.Web.WebView2.Core) | Microsoft Learn](https://learn.microsoft.com/en-us/dotnet/api/microsoft.web.webview2.core.corewebview2.postwebmessageasjson?view=webview2-dotnet-1.0.1418.22)
CoreWebView2.PostWebMessageAsJson(String) Method (Microsoft.Web.WebView2.Core)
Posts the specified webMessageAsJson to the top level document in this WebView. 


window.chrome.webview.addEventListener('message', handler)
window.chrome.webview.removeEventListener('message', handler)